rpcs3/Emu/Cell/lv2/sys_usbd.cpp

```diff
@@ -861,8 +861,15 @@ error_code sys_usbd_get_descriptor(ppu_thread& ppu, u32 handle, u32 device_handl
 		return CELL_EINVAL;
 	}
 
-	u8* ptr = static_cast<u8*>(descriptor.get_ptr());
+	if (!descriptor)
+	{
+		return CELL_EFAULT;
+	}
+
+	std::vector<u8> desc_buf(usbh.handled_devices[device_handle].second->device.get_size());
```

Comment:
Why do you use a temporary?

Reply:
Because we cannot specify a size for `usbh.handled_devices[device_handle].second->device.write_data(ptr);` if `ptr` points to `descriptor.get_ptr()` directly.

Reply:
Cant the method write_data be rewritten to accept a size?

Reply:
Sure thing.
Please validate.